### PR TITLE
Feature/day 07

### PR DESCRIPTION
**Pull request by Jaxolotl**

### CODE REVIEW

File: lessons/day_07/day_07.md [hunk 1]
```diff
@@ -8,23 +8,366 @@
...
+      { "name": "Radagast" },
+      { "name": "Galadriel" }
+    ]
+  }
+}
+```
+
+### 4. and in your mind bind them — **"errors"** as **"unrequested results"**
+
+Let's stay put for a moment and analyse what just happened above.
```
Comment (ealvarezk) [inline lessons/day_07/day_07.md:215]: you could change analyse to analyze

File: lessons/day_07/day_07.md [hunk 2]
```diff
@@ -8,23 +8,366 @@
...
+
+1. One of our records doesn't respect the contract for it's type
```
Comment (ealvarezk) [inline lessons/day_07/day_07.md:217]: you could change it's to its

File: lessons/day_07/day_07.md [hunk 3]
```diff
@@ -8,23 +8,366 @@
...
+
+So:
+
+1. Should we consider the whole as an error?
+2. Should we tell the user something about the fault? what? how? when?
+3. Can/Should the user do anything about it?
+
+... and many other questions can derive from this behavior!  
+
+So what do we do about this?
```
Comment (ealvarezk) [inline lessons/day_07/day_07.md:229]: You can add a comma after So

File: lessons/day_07/day_07.md [hunk 4]
```diff
@@ -8,23 +8,366 @@
...
+— Leverage your GraphQL server app error features!  
+— Do Not! ୧༼ಠ益ಠ༽୨
+
+— Be careful with `extension` field!  
+— Hell yeah! ᕦ( ͡° ͜ʖ ͡°)ᕤ
+
+— Disable `stacktrace` for production! _(should be the default behaviour of your server app in production mode, but still...)_  
+— Hell yeah! ᕦ( ͡° ͜ʖ ͡°)ᕤ
+
+Despite the last two assertions we all agree, unless the only thing you have is a fatal error you'll need to start thinking of **"errors"** in terms of **results**, which is true in the GraphQL mindset even though they might not be the results you're expecting for. You may think of them as **requested** and **unrequested** results and **make sure all the organization is aligned on how to organize and treat them**.
```
Comment (ealvarezk) [inline lessons/day_07/day_07.md:252]: I don't find it clear

File: lessons/day_07/day_07.md [hunk 5]
```diff
@@ -8,23 +8,366 @@
...
+— Make Errors part of your schema!  
+— Do Not! ୧༼ಠ益ಠ༽୨
+
+— Leverage your GraphQL server app error features!  
+— Do Not! ୧༼ಠ益ಠ༽୨
+
+— Be careful with `extension` field!  
+— Hell yeah! ᕦ( ͡° ͜ʖ ͡°)ᕤ
+
+— Disable `stacktrace` for production! _(should be the default behaviour of your server app in production mode, but still...)_
```
Comment (ealvarezk) [inline lessons/day_07/day_07.md:249]: you can change behaviour with behavior

File: lessons/day_07/day_07.md [hunk 6]
```diff
@@ -8,23 +8,366 @@
...
+2. [**One Place** to find them](#2-one-place-to-find-them--200-ok--error)  
+_`200 OK` & `error`_
+3. [**One Object** to bring them all](#3-one-object-to-bring-them-all--data--error-are-siblings)  
+_`{ "error": {...}, "data": {...}}`_
+4. [and in your mind **bind them**](#4-and-in-your-mind-bind-them--errors-as-unrequested-results)  
+_"errors" as "unrequested results"_
+5. [in the land of the **runtime** where the **implementation** lies.](#5-in-the-land-of-the-runtime-where-the-implementation-lies--the-runtime-handles-the-rest)  
+_The runtime handles the rest._
+
+### 1. One Graph to rule the all — The Spec.
```
Comment (quequitejerina) [inline lessons/day_07/day_07.md:26]: Change "the" with "them"

File: lessons/day_07/day_07.md (hunk 1 above)
Comment (Jaxolotl) [reply lessons/day_07/day_07.md:215]: both are valid

File: lessons/day_07/day_07.md (hunk 5 above)
Comment (Jaxolotl) [reply lessons/day_07/day_07.md:249]: both are valid

File: lessons/day_07/day_07.md [hunk 7]
```diff
@@ -8,23 +8,366 @@
...
+So:
+
+1. Should we consider the whole as an error?
+2. Should we tell the user something about the fault? what? how? when?
+3. Can/Should the user do anything about it?
+
+... and many other questions can derive from this behavior!  
+
+So what do we do about this?  
+There's no "the right answer" for that.
```
Comment (quequitejerina) [inline lessons/day_07/day_07.md:230]: Would it be better without the "the"? "_There´s no 'right answer' for that_"

File: lessons/day_07/day_07.md [hunk 8]
```diff
@@ -8,23 +8,366 @@
...
+1. Should we consider the whole as an error?
+2. Should we tell the user something about the fault? what? how? when?
+3. Can/Should the user do anything about it?
+
+... and many other questions can derive from this behavior!  
+
+So what do we do about this?  
+There's no "the right answer" for that.  
+
+In terms of best practices' community trends there are different opinions:
```
[the remark this answers is not in the document]
Comment (Jaxolotl) [reply lessons/day_07/day_07.md:232]: Will rephrase, it wasn't that but obviously I didn't expressed myself clearly

File: lessons/day_07/day_07.md (hunk 7 above)
Comment (Jaxolotl) [reply lessons/day_07/day_07.md:230]: it refers to the sentence "The right answer" as a superlative assertion. There might be "a right answer" taking in consideration the context, but it'll never be "THE" right answer.

Comment (quequitejerina) [reply lessons/day_07/day_07.md:230]: Yes, i know the sentence, but given the context i think just "right answer" will suffice. it is up to you 😉

**Pull request merged**